lib/endpoints/class-wp-json-comments-controller.php

```diff
@@ -418,37 +419,68 @@ public function prepare_item_for_response( $comment, $request ) {
 		);
 
 		if ( 'edit' == $request['context'] ) {
-			$fields['author']['email']      = $comment->comment_author_email;
-			$fields['author']['ip']         = $comment->comment_author_IP;
-			$fields['author']['user_agent'] = $comment->comment_agent;
-			$fields['date_gmt']             = json_mysql_to_rfc3339( $comment->comment_date_gmt );
-			$fields['content']['raw']       = $comment->comment_content;
-			$fields['karma']                = $comment->comment_karma;
+			$data['author']['email']      = $comment->comment_author_email;
+			$data['author']['ip']         = $comment->comment_author_IP;
+			$data['author']['user_agent'] = $comment->comment_agent;
+			$data['date_gmt']             = json_mysql_to_rfc3339( $comment->comment_date_gmt );
+			$data['content']['raw']       = $comment->comment_content;
+			$data['karma']                = $comment->comment_karma;
 		}
 
-		$links = array();
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object
+		$data = json_ensure_response( $data );
+
+		$links = $this->prepare_links( $comment );
+		foreach ( $links as $rel => $attributes ) {
+			$other = $attributes;
+			unset( $other['href'] );
+			$data->add_link( $rel, $attributes['href'], $other );
+		}
```

Comment:
We can probably just ignore `href` inside `add_link`, which would shrink this down to a single loop instead. Maybe add a `add_links` too. Will change.